mock/environment/__init__.py

```diff
@@ -16,66 +16,120 @@
-                    night_list.append(current_line)
-                    previous_line = current_line
-                    try:
-                        index3, current_line = next(local_site_data)
-                        current_line["cc_band"] = cc_band_to_float(current_line["cc_band"])
-                    except StopIteration:
-                        logging.info("End of data")
-                        break
-               
-                self.site_data_by_night[site][night_date] = night_list
-
+                input_data = input_data.drop(columns={
+                    'Night', f'{site.name}_Sun_Elevation', 'cc_requested', 'iq_requested',
+                    'Airmass_QAP', 'Filter', 'Instrument', 'Object', 'Sidereal', 'Filename_QAP', 'Types',
+                    'adaptive_optics', 'calibration_program', 'cass_rotator_pa', 'data_label', 'dec',
+                    'engineering', 'exposure_time', 'filename', 'local_time', 'object', 'observation_class',
+                    'observation_id', 'observation_type', 'program_id', 'ra', 'science_verification',
+                    'types', 'wavefront_sensor', 'RA', 'DEC', 'Azimuth', 'Elevation', 'Airmass', 'Filename_FITS',
+                    'IQ_OIWFS', 'IQ_P2WFS', 'IQ_OIWFS_MEDIAN', 'IQ_P2WFS_MEDIAN', 'QAP_IQ', 'QAP_IQ_WFS_scaled',
+                    'Ratio_OIWFS', 'Ratio_P2WFS', 'QAP_IQ_WFS_scaled500', 'IQ_P2WFS_Zenith', 'IQ_OIWFS_Zenith',
+                    'IQ_P2WFS_MEDIAN_Zenith', 'QAP_IQ_WFS_scaled500_Zenith'})
+
+                self.site_data_by_night[site] = {}
+                local_site_data = input_data.iterrows()
+                for index, night in local_site_data:
+                    night_start_line = night
+                    night_date = night[Env._time_stamp].date()
+                    night_start_line["cc_band"] = Env._cc_band_to_float(night_start_line["cc_band"])
+                    night_start_line["iq_band"] /= 100
+                    logging.info(f'\tProccesing UTC night of {night_date}')
+                    night_list = [night_start_line]
+                    previous_line = night_start_line
+                    index2, current_line = next(local_site_data)
+                    current_line["cc_band"] = Env._cc_band_to_float(current_line["cc_band"])
+                    current_line["iq_band"] /= 100
+
+                    while current_line[Env._time_stamp] - previous_line[Env._time_stamp] < Env._day_difference:
+                        night_list.append(current_line)
+                        previous_line = current_line
+                        try:
+                            index3, current_line = next(local_site_data)
+                            current_line["cc_band"] = Env._cc_band_to_float(current_line["cc_band"])
+                            current_line["iq_band"] /= 100
+                        except StopIteration:
+                            logging.info("End of data")
+                            break
```

Comment:
I have some concerns about this not working properly because you read in from the iterator three times and there's no guarantee that there will be three rows to read.

It could also be cleaned up since you do almost exactly the same thing three times (so it should be extracted to a function).

Let's worry about that later and make sure this works.